src/api.js

```diff
@@ -1,39 +1,50 @@
-var ajax  = require('./ajax')
-var route = require('./route')
-var resource = require('./resource')
-var url = require('./url')
+var ajax      = require('./ajax')
+var assign    = require('./assign')
+var prepare   = require('./prepare')
+var resource  = require('./resource')
+var route     = require('./route')
+var url       = require('./url')
+var segmentize = require('./segmentize')
+
+function API (config, routes) {
+  if (this instanceof API === false) {
+    return new API(config, routes)
+  }
 
-module.exports = function (config, routes) {
+  this.config   = prepare(config)
+  this.segments = []
 
-  if (!config) {
-    throw new TypeError('Please provide a configuration object as the first argument.')
-  }
+  this.route(routes)
+}
 
-  if ('baseURL' in config === false) {
-    throw new TypeError('baseURL configuration option is required')
-  }
+API.prototype = {
+  ajax: ajax,
 
-  var API = {
-    config: config,
-    ajax: ajax,
+  namespace: function (key) {
+    var child = Object.create(this, {
+      segments: { value: this.segments.concat(key) }
+    })
 
-    toString: function () {
-      return config.baseURL
-    },
+    this[key] = key in this ? assign(child, this[key]) : child
 
-    resolve: function (path, params) {
-      return url(config.baseURL, path, params)
-    },
+    return child
```

Comment:
`child` is a funny name to me here. That or I'm not understanding what's going on in this method.